src/ui/StartUp.java

```diff
@@ -104,7 +104,7 @@ public static void main(String[] args) {
 			frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
 			frame.setTitle("Cognitive Bias Modifier");
 			frame.setResizable(false);
-			frame.add(new StartUp(4, 750, 30, new ImageManager()), BorderLayout.CENTER);
+			frame.add(new StartUp(4, 750, 30, new ImageManager(new int[12])), BorderLayout.CENTER);
```

Comment:
Uita-te peste metoda _initialize_. O sa vezi ca in interiorul ei se face o asignare la campul ```this.images```. Poate ca nu mai e nevoie de ea, mai ales acum dupa aceste modificari.

Reply:
Da, am vazut, doar ca ma gandeam cum sa fac sa fie codul mai _maintainable_, deoarece in declaratia initiala: this.images, images lua valoarea la numberOfImages, care la randul ei putea fi usor schimbata prin simpla schimbare  a parametrului numberOfColumns la instantierea obiectului StrartUp